src/token/erc721/extensions/erc721_uri_storage.cairo

```diff
@@ -0,0 +1,102 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts for Cairo v0.14.0 (token/erc721/extensions/erc721_uri_storage.cairo)
+
+/// # ERC721URIStorage Component
+///
+/// The ERC721URIStorage component provides a flexible IERC721Metadata implementation that enables
+/// storage-based token URI management.
+#[starknet::component]
+pub mod ERC721URIStorageComponent {
+    use openzeppelin::introspection::src5::SRC5Component;
+    use openzeppelin::token::erc721::ERC721Component::InternalImpl as ERC721Impl;
+    use openzeppelin::token::erc721::interface::IERC721Metadata;
+    use openzeppelin::token::erc721::{ERC721Component, ERC721HooksEmptyImpl};
+    use starknet::ContractAddress;
+
+    #[storage]
+    struct Storage {
+        ERC721URIStorage_token_uris: LegacyMap<u256, ByteArray>,
```

Comment:
@andrew-fleming, the tests are failing and `Map` is not identified in the tests? Can you please guide on this?

Reply:
> the tests are failing and Map is not identified in the tests? Can you please guide on this?

You need to bring `starknet::storage::Map` into scope

> Is this because my local version of scarb is not updated to the one used in the repository?

If you're trying to compile and run tests locally, yes. The scarb version should be the same.

> But I am wondering that the tests are running as a workflow should not fail

The CI uses the scarb version set in the toml

Reply:
@andrew-fleming, updated that. Thanks